File: backend/entity/report.go
package entity


import (

   "time"

   "gorm.io/gorm"

)

type Report struct {

   gorm.Model

   Note string    `json:"note"`

   Contact		string  `json:"contact"`

   Photo string    `gorm:"type:longtext"`

   DateApprove  time.Time `json:"dapprove"`

   DateReport  time.Time `json:"dreport"`

   UsersID  uint      `json:"users_id"`

   Status  string      `json:"status"`

   Users    *Student  `gorm:"foreignKey: users_id" json:"users"`

   AdminID  uint      `json:"admin_id"`

   Admin    *Admin  `gorm:"foreignKey: admin_id" json:"admin"`

   BooksID  uint      `json:"books_id"`

   Books    *Books  `gorm:"foreignKey: books_id" json:"book"`

   DormitoryID  uint      `json:"dormitory_id"`

   Dormitory    *Dormitory  `gorm:"foreignKey: dormitory_id" json:"dorm"`

   RoomID  uint      `json:"room_id"`

   Room    *Room  `gorm:"foreignKey: room_id" json:"room"`

}

File: frontend/src/interfaces/Report.ts
export interface ReportInterface {
    dreport: string | number | Date | null | undefined;
    dapprove: string | number  | Date | null | undefined;
    users: any;
    room: any;
    dorm: any;

    ID?: number;
  
    note?: string;

    contact?: string;

    Photo?: string;

    DateReport?: string;

    DateApprove?: string;
  
    status?: string;
  
    admin_id?: number;

    users_id?: number;

    dormitory_id?: number;

    books_id?: number;

    room_id?: number;
  
  }